app/src/main/java/org/sil/storyproducer/controller/BaseController.kt

```diff
@@ -79,11 +79,15 @@ open class BaseController(
     }
 
     private fun onStoriesUpdated() {
-        if (Workspace.registration.complete) {
-            view.showMain()
-        } else {
-            view.showRegistration()
+        if (!Workspace.registration.complete) {
+            // DKH - 05/12/2021
+            // Issue #573: SP will hang/crash when submitting registration
+            // Tell MainActivity to display the registration upon MainActivity startup
```

Comment:
how about- Don't call view.showRegistration() directly, instead tell MainActivity to display the registration upon MainActivity startup

Reply:
done